crates/ruff_python_literal/src/format.rs

```diff
@@ -320,41 +329,61 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
 
 /// Parses a format part within a format spec
 fn parse_nested_placeholder<'a>(
-    parts: &mut Vec<FormatPart>,
+    placeholders: &mut Vec<FormatPart>,
     text: &'a str,
 ) -> Result<&'a str, FormatSpecError> {
     match FormatString::parse_spec(text, AllowPlaceholderNesting::No) {
-        // Not a nested replacement, OK
+        // Not a nested placeholder, OK
         Err(FormatParseError::MissingStartBracket) => Ok(text),
         Err(err) => Err(FormatSpecError::InvalidPlaceholder(err)),
         Ok((format_part, text)) => {
-            parts.push(format_part);
+            placeholders.push(format_part);
             Ok(text)
         }
     }
 }
 
+/// Parse and consume all placeholders in a format spec
+/// This will also consume any intermediate characters such as `x` and `y` in `"x{foo}y{bar}z"`
+/// If no placeholders are present, the text will be returned unmodified.
+fn consume_all_placeholders<'a>(
+    placeholders: &mut Vec<FormatPart>,
+    text: &'a str,
+) -> Result<&'a str, FormatSpecError> {
+    let mut chars = text.chars();
+    let mut text = text;
+    let mut placeholder_count = placeholders.len();
+
+    while chars.clone().contains(&'{') {
+        text = parse_nested_placeholder(placeholders, chars.as_str())?;
+        chars = text.chars();
+        // If we did not parse a placeholder, consume a character
+        if placeholder_count == placeholders.len() {
+            chars.next();
+        } else {
+            placeholder_count = placeholders.len();
+        }
+    }
+    Ok(text)
+}
```

Comment:
Since you mentioned that this felt awkward to you in the prior PR, I tried to simplify it a little by using the text directly:

```rust
/// Parses a format part within a format spec
fn parse_nested_placeholder(text: &str) -> Result<Option<(FormatPart, &str)>, FormatSpecError> {
    match FormatString::parse_spec(text, AllowPlaceholderNesting::No) {
        // Not a nested placeholder, OK
        Err(FormatParseError::MissingStartBracket) => Ok(None),
        Err(err) => Err(FormatSpecError::InvalidPlaceholder(err)),
        Ok((format_part, text)) => Ok(Some((format_part, text))),
    }
}

/// Parse and consume all placeholders in a format spec
/// This will also consume any intermediate characters such as `x` and `y` in `"x{foo}y{bar}z"`
/// If no placeholders are present, the text will be returned unmodified.
fn consume_all_placeholders(mut text: &str) -> Result<Vec<FormatPart>, FormatSpecError> {
    let mut placeholders = vec![];
    while let Some(bracket) = text.find('{') {
        if let Some((format_part, rest)) = parse_nested_placeholder(&text[bracket..])? {
            text = rest;
            placeholders.push(format_part);
        } else {
            text = &text[bracket + 1..];
        }
    }
    Ok(placeholders)
}
```

Don't feel strongly, only a suggestion.

Reply:
I'll study this :) thank you!